orchagent/neighorch.cpp

```diff
@@ -1227,6 +1260,185 @@ bool NeighOrch::removeNeighbor(const NeighborEntry &neighborEntry, bool disable)
+                           macAddress.to_string().c_str(), alias.c_str(), sai_serialize_status(status).c_str());
+                task_process_status handle_status = handleSaiCreateStatus(SAI_API_NEIGHBOR, status);
+                if (handle_status != task_success)
+                {
+                    return parseHandleSaiStatusFailure(handle_status);
+                }
+            }
+        }
+
+        SWSS_LOG_NOTICE("Created neighbor ip %s, %s on %s", ip_address.to_string().c_str(),
+                macAddress.to_string().c_str(), alias.c_str());
+
+        m_intfsOrch->increaseRouterIntfsRefCount(alias);
+
+        if (neighbor_entry.ip_address.addr_family == SAI_IP_ADDR_FAMILY_IPV4)
+        {
+            gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV4_NEIGHBOR);
+        }
+        else
+        {
+            gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV6_NEIGHBOR);
+        }
+
+        if (!addNextHop(NextHopKey(ip_address, alias)))
+        {
+            status = sai_neighbor_api->remove_neighbor_entry(&neighbor_entry);
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to remove neighbor %s on %s, rv:%d",
+                               macAddress.to_string().c_str(), alias.c_str(), status);
+                task_process_status handle_status = handleSaiRemoveStatus(SAI_API_NEIGHBOR, status);
+                if (handle_status != task_success)
+                {
+                    return parseHandleSaiStatusFailure(handle_status);
+                }
+            }
+            m_intfsOrch->decreaseRouterIntfsRefCount(alias);
+
+            if (neighbor_entry.ip_address.addr_family == SAI_IP_ADDR_FAMILY_IPV4)
+            {
+                gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_IPV4_NEIGHBOR);
+            }
+            else
+            {
+                gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_IPV6_NEIGHBOR);
+            }
+
+            return false;
+        }
+    }
+
+    m_syncdNeighbors[neighborEntry] = { macAddress, true };
+
+    NeighborUpdate update = { neighborEntry, macAddress, true };
+    notify(SUBJECT_TYPE_NEIGH_CHANGE, static_cast<void *>(&update));
+
+    return true;
+}
+
+/* Process bulk ctx entry and disable the neigbor */
+bool NeighOrch::processBulkDisableNeighbor(NeighborContext& ctx)
+{
+    SWSS_LOG_ENTER();
+
+    const auto& object_statuses = ctx.object_statuses;
+    auto it_status = object_statuses.begin();
+    sai_status_t status;
+
+    const NeighborEntry neighborEntry = ctx.neighborEntry;
+    string alias = neighborEntry.alias;
+    IpAddress ip_address = neighborEntry.ip_address;
+
+    if (m_syncdNeighbors.find(neighborEntry) == m_syncdNeighbors.end())
+    {
+        return true;
+    }
+
+    SWSS_LOG_INFO("Checking neighbor remove entry status %s on %s.", ip_address.to_string().c_str(), m_syncdNeighbors[neighborEntry].mac.to_string().c_str());
+
+    if (isHwConfigured(neighborEntry))
```

Comment:
Took a loot at this, we still need the HwConfigured check since this isn't checked for in the disableNeighbors patch